File: app/src/main/java/com/example/mvcexempel/MainActivity.java
package com.example.mvcexempel;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.Toast;

import java.util.List;

public class MainActivity extends AppCompatActivity implements MainContract.View, MemberContract.View {

    MainContract.Presenter presenter;
    MemberContract.Presenter memberPresenter;

    EditText userNameInput;
    EditText userPasswordInput;
    ListView memberListView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        presenter = new MainPresenter(new DataManager(), this, new Navigator(this));
        memberPresenter = new MemberPresenter(new DataManager(), this);

        memberListView = findViewById(R.id.lv_members);

        userNameInput = findViewById(R.id.et_name);
        userPasswordInput = findViewById(R.id.et_password);
        Button loginButton = findViewById(R.id.btn_login);
        Button registerButton = findViewById(R.id.btn_register);


        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onRegisterButtonClicked();
                memberPresenter.onViewCreated();

            }
        });

        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.onLoginButtonClicked();
            }
        });

        presenter.onViewCreated();
    }

    @Override
    public void showLoginSuccessMessage() {
        Toast.makeText(this, "Login successful", Toast.LENGTH_LONG).show();
    }

    @Override
    public void showLoginFaildMessage() {
        Toast.makeText(this, "wrong username/ password", Toast.LENGTH_LONG).show();
    }

    @Override
    public void showRegisteredSuccessMessage(User user) {
        String s ="user " + user.getUserName() + " registered";
        Toast.makeText(this,s , Toast.LENGTH_LONG).show();
    }

    @Override
    public void showRegisteredFailedMessage() {
        Toast.makeText(this, "registration failed", Toast.LENGTH_LONG).show();
    }

    @Override
    public String getUserName() {
        return userNameInput.getText().toString();
    }

    @Override
    public String getUserPassword() {
        return userPasswordInput.getText().toString();
    }

    @Override
    public void displayMembers(List<User> members) {
        MemberAdapter adapter = new MemberAdapter(this,members);
        memberListView.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/kotlinklasserexempel/MainActivity.kt
package com.example.kotlinklasserexempel

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.kotlinklasserexempel.databinding.ActivityMainBinding
import java.io.DataInput

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val user = DataKlass("Bill",42,"mail.com","0731234567")


        binding.button.setOnClickListener {
            val intent = Intent(this,NewActivity::class.java)
            intent.putExtra("currentUser", user)
            startActivity(intent)
        }

//        val user = User("Bosse", 42)

//        val dataUser =DataKlass("Bill", 42, "mail.com", "+4670012345")
//        val dataUser2 = dataUser.copy(name = "nån annan")

//        println("user = ${user}")
//        println("dataUser = ${dataUser}")
//        println("dataUser2 = ${dataUser2}")





    }
}